src/core/AutoRest/Plugins/AzureValidator.cs

```diff
@@ -82,7 +83,12 @@ private void LogValidationMessage(ValidationMessage validationMessage)
 
         var serviceDefinition = SwaggerParser.Load(files[0], fs);
         var validator = new RecursiveObjectValidator(PropertyNameResolver.JsonName);
-        foreach (ValidationMessage validationEx in validator.GetValidationExceptions(new Uri(files[0], UriKind.RelativeOrAbsolute), serviceDefinition))
+        var metadata = new ServiceDefinitionMetadata
+            {
+                ServiceDefinitionDocumentType = (ServiceDefinitionDocumentType)Enum.Parse(typeof(ServiceDefinitionDocumentType), (await GetValue("openapi-type"))?.ToString().ToUpper() ?? ServiceDefinitionDocumentType.ARM.ToString()),
+                MergeState = await GetValue<bool?>("is-individual-swagger") == true ? ServiceDefinitionMergeState.Before : ServiceDefinitionMergeState.After
```

Comment:
I have added explanation under ```ServiceDefinitionMetadata``` is that not sufficient? Not sure I understand the latter part of the comment.